### ISSUE TITLE
Glob matching does not work when specified in script using Yarn 2

### ISSUE THREAD

**Issue by Robin-Hoodie** [mysticatea/npm-run-all#200]
Running a script `echo: run-p echo:*` defined in `package.json`  will not work on Yarn 2.

Example `package.json`:

```
{
  "name": "test-yarn1",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "echo:bar": "echo 'bar'",
    "echo:foo": "echo 'foo'",
    "echo": "run-p echo:*"
  },
  "devDependencies": {
    "npm-run-all": "^4.1.5"
  }
}
```

To reproduce:

- Create a dir
- Run `yarn init -y`
- Run `yarn add -D npm-run-all`
- Add 3 scripts to `package.json`:
  - `"echo:foo":  "echo 'foo'"`
  - `"echo:bar": "echo 'bar'"`
  -  `"echo": "run-p echo:*"`
- Run `yarn set version berry`
- Run `yarn echo`

=> No matches found: "echo:*"

Oddly enough, running `yarn run-p echo:*` directly does still work

Reproduction repo: https://github.com/Robin-Hoodie/yarn2-npm-run-all-repro

**Comment by nigelzor** [comment]
I just ran into this myself. If you quote the `*` then yarn won't try to handle the wildcard and `run -p` will work as expected.

Try this:
```
  "scripts": {
    "echo:bar": "echo 'bar'",
    "echo:foo": "echo 'foo'",
    "echo": "run-p 'echo:*'"
  }
```

**Comment by Robin-Hoodie** [comment]
Thanks for the tip, that works!

I do think the root cause of this should be fixed however (whether here or in Yarn 2) 
If not that, at minimum some documentation can be added

**Comment by arthurfiorette** [comment]
I ended up falling for that too. Please add it to the documentation

**Comment by BasixKOR** [comment]
I don't think this is a bug for both sides. Please don't get mad at me; I was also confused by the error. However, `*` is a valid shell syntax for globs, and Yarn implements a bash-like shell that presumably expands the wildcard.

It seems like expanding the wildcard is the expected behaviour of Yarn, and it doesn't look like `npm-run-all` can solve it either.

**Comment by waptik** [comment]
> I just ran into this myself. If you quote the `*` then yarn won't try to handle the wildcard and `run -p` will work as expected.
> 
> Try this:
> 
> ```
>   "scripts": {
>     "echo:bar": "echo 'bar'",
>     "echo:foo": "echo 'foo'",
>     "echo": "run-p 'echo:*'"
>   }
> ```

Thanks very much for this. I had to line up all related script commands on one line as I didn't know what to do.

**Comment by mathieutu** [comment]
Hey, just had the issue too.
 
>it doesn't look like npm-run-all can solve it either.

Agree, but adding it to the doc could be a good help, it avoids people to look for this issue 🙂.

**Comment by BensThoughts** [comment]
Ditto to all of the above.  Thank goodness for github issues and threads like these.  This was the first thing that broke when I updated to yarn 2, given that the docs say it will work with yarn this was a little confusing to find the solution here.

**Comment by kachkaev** [comment]
Adding double quotes works for me in Yarn 4.0.0-rc.27:

```diff
  "scripts": {
    "echo:bar": "echo \"bar\"",
    "echo:foo": "echo \"foo\"",
-   "echo": "npm-run-all echo:*"
+   "echo": "npm-run-all \"echo:*\""
  }
```

Otherwise, it says `No matches found: "echo:*"`. Using double quotes for compatibility with `cmd.exe`.

**Comment by Thanaen** [comment]
I find it odd that this works for all the other manager packages (npm, yarn classic, pnpm, bun), but not for modern Yarn. 😅
